services/web/server/src/simcore_service_webserver/projects/lock.py

```diff
@@ -49,10 +51,13 @@ async def lock_project(
     )
     try:
         if not await redis_lock.acquire(
-            blocking=False,
+            blocking_timeout=PROJECT_LOCK_ACQUIRE_TIMEOUT,
```

Comment:
So if I get your "fix" right, instead of raising right away, it will now try for 5 seconds before raising? is this correct? I think this is maybe not too bad for all the other calls to ```lock_project```. Please note that this is used not only for cloning, but for opening, closing, (exporting).

Now the question is more why 5 seconds? is this due to @wvangeit number of calls? what will then happen if he multiplies again by 10 the number of projects? To me this looks very much like if I may "let's add a sleep of a few seconds in the test". Maybe now we are touching something that was made to work with the frontend (where we must ensure that the project is not edited while being cloned), that is maybe not so good to use directly from the public API?
Some questions:
- is this project you are cloning edited by the frontend?
- can someone else than the current owner edit that project while you are cloning?
- should we think of some kind of read only locking mechanism? where several can acquire the lock as long as they only want to copy?

Reply:
I completely agree, that this fix is not a proper fix. It was the first thing that came into my mind when I was looking at this in order to get @wvangeit's demo to run. The 5sec was picked out of the blue air. I have not yet tested if this is even enough to make the demo run. 

Over the weekend I thought more about a solution similar to the third suggestion you make. If we are really only cloning here I guess the only issue one could run into is that the project is modified while the clone happens. Another solution could be to create a "batch clone" endpoint which allows to clone up to `N` copies at a time. Let's discuss face to face in the office.